cloudpickle/cloudpickle.py

```diff
@@ -282,6 +282,21 @@ def save_function(self, obj, name=None):
             self.memoize(obj)
     dispatch[types.FunctionType] = save_function
 
+    def _save_subimports(self, code, top_level_dependencies):
+        """Ensure de-pickler imports any package sub-modules needed by the function"""
+        # check if any known dependency is an imported package
+        for x in top_level_dependencies:
+            if isinstance(x, types.ModuleType) and x.__package__:
+                # check if the package has any currently loaded sub-imports
+                prefix = x.__name__ + '.'
+                for name, module in sys.modules.items():
+                    if name.startswith(prefix):
+                        # check whether the function can address the sub-module
+                        tokens = set(name[len(prefix):].split('.'))
+                        if not tokens - set(code.co_names):
+                            self.save(module) # ensure the unpickler executes import of this submodule
+                            self.write(pickle.POP) # then discard the reference to it
```

Comment:
cosmetic: could you please move the comments to be on their own lines (before the matching statements) so as to avoid long (80+ columns lines).